### PR TITLE
Improving descriptions in OpenAPI

### PR DESCRIPTION
**Pull request by mik-laj**
Preview:  https://redocly.github.io/redoc/?url=https://github.com/PolideaInternal/airflow/raw/api-description/airflow/api_connexion/openapi/v1.yaml

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: airflow/api_connexion/openapi/v1.yaml [hunk 1]
```diff
@@ -1256,65 +1319,75 @@ components:
...
       properties:
         import_error_id:
           type: integer
           readOnly: true
+          description: The import error ID
         timestamp:
           type: string
           format: datetime
           readOnly: true
+          description: The time when this error was created,
```
Comment (michalslowikowski00) [inline airflow/api_connexion/openapi/v1.yaml]:
```suggestion
          description: The time when this error was created
```

File: airflow/api_connexion/openapi/v1.yaml [hunk 2]
```diff
@@ -1324,59 +1397,81 @@ components:
...
-          readOnly: true
-          nullable: true
 
     SLAMiss:
       type: object
       properties:
         task_id:
           type: string
           readOnly: true
+          description: The Task ID
```
Comment (michalslowikowski00) [inline airflow/api_connexion/openapi/v1.yaml]:
```suggestion
          description: The Task ID.
```

Comment (michalslowikowski00) [reply airflow/api_connexion/openapi/v1.yaml]: Sometimes there is full stop at the end of the sentence and sometimes isn't.

File: airflow/api_connexion/openapi/v1.yaml [hunk 3]
```diff
@@ -1052,17 +1064,40 @@ paths:
...
           $ref: '#/components/responses/Unauthenticated'
         '403':
           $ref: '#/components/responses/PermissionDenied'
 
   /health:
     get:
-      summary: Returns the status of Airflow's metadatabase and scheduler
+      summary: Get a instance status
+      description: |
+        Get the status of Airflow's metadatabase and scheduler.
```
Comment (houqp) [inline airflow/api_connexion/openapi/v1.yaml]: I feel like we can just use `Get the health status of Airflow's metadatabase and scheduler` as summary here and remove the description key. `Get an instance status` doesn't really mean anything from an external reader's point of view.

File: airflow/api_connexion/openapi/v1.yaml [hunk 4]
```diff
@@ -1076,7 +1111,7 @@ paths:
 
   /version:
     get:
-      summary: Get version information
+      summary: Get a version information
```
Comment (houqp) [inline airflow/api_connexion/openapi/v1.yaml:1115]: would be better to just say `Get current Airflow version` here. It's unclear what `version information` actually means.

in the response section below, we have the description set to `Return current configuration`. Is this still accurate?

File: airflow/api_connexion/openapi/v1.yaml (hunk 3 above)
Comment (mik-laj) [reply airflow/api_connexion/openapi/v1.yaml]: The summary should be no longer than 20 characters to make it accessible to all people. The description can be of any length. I don't think the problem is that the summary doesn't say everything as users can access the full description easily, but the short titles make it easier to navigate.

Screenshot 2020-08-23 at 08 56 27
https://airflow.readthedocs.io/en/latest/stable-rest-api-ref.html#tag/Monitoring

File: airflow/api_connexion/openapi/v1.yaml (hunk 4 above)
Comment (mik-laj) [reply airflow/api_connexion/openapi/v1.yaml:1115]: This endpoint is in a separate section, so in my opinion it expresses its intention clearly enough. The summary should also not be longer than 20 characters to be accessible to everyone.

I will correct the description for the succesfull response.

File: airflow/api_connexion/openapi/v1.yaml [hunk 5]
```diff
@@ -88,13 +88,13 @@ paths:
...
     get:
-      summary: Get a connection entry
+      summary: Get a connection
       x-openapi-router-controller: airflow.api_connexion.endpoints.connection_endpoint
       operationId: get_connection
       tags: [Connection]
       responses:
         '200':
-          description: Successful response.
+          description: Sucess.
```
Comment (turbaszek) [inline airflow/api_connexion/openapi/v1.yaml:97]: In L49 we use more meaningful description. Should we unify the style?

Comment (mik-laj) [reply airflow/api_connexion/openapi/v1.yaml:97]: Right. This field should describe **when** this response is returned and not contain information on what the field contains. We have other places to describe the content.

Comment (mik-laj) [reply airflow/api_connexion/openapi/v1.yaml:97]: I update the description.

**Review by turbaszek — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Pull request merged**